Code/Framework/AzFramework/Tests/DocumentPropertyEditor/CvarAdapterTests.cpp

```diff
@@ -0,0 +1,170 @@
+    AZ_CVAR(bool, dpe_TestBool, false, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "DPE test bool");
+    AZ_CVAR(AZ::Vector2, dpe_TestVec2, AZ::Vector2::CreateZero(), nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "DPE test vec2");
+    AZ_CVAR(AZ::Vector3, dpe_TestVec3, AZ::Vector3::CreateOne(), nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "DPE test vec3");
+    AZ_CVAR(AZ::Color, dpe_TestColor, AZ::Color(), nullptr, ConsoleFunctorFlags::DontReplicate, "DPE test color");
+
+    class CvarAdapterDpeTests : public DocumentPropertyEditorTestFixture
+    {
+    public:
+        void SetUp() override
+        {
+            DocumentPropertyEditorTestFixture::SetUp();
+            m_console = AZStd::make_unique<AZ::Console>();
+            m_console->LinkDeferredFunctors(AZ::ConsoleFunctorBase::GetDeferredHead());
+            AZ::Interface<AZ::IConsole>::Register(m_console.get());
+            m_adapter = AZStd::make_unique<CvarAdapter>();
+        }
+
+        void TearDown() override
+        {
+            m_adapter.reset();
+            AZ::Interface<AZ::IConsole>::Unregister(m_console.get());
+            m_console.reset();
+            DocumentPropertyEditorTestFixture::TearDown();
+        }
+
+        Dom::Value GetEntryRow(AZStd::string_view cvarName)
+        {
+            Dom::Value rows = m_adapter->GetContents();
+            for (auto it = rows.ArrayBegin(); it != rows.ArrayEnd(); ++it)
+            {
+                Dom::Value label = (*it)[0].GetNodeValue();
+                if (label.GetString() == cvarName)
+                {
+                    return *it;
+                }
+            }
+            return {};
+        }
+
+        Dom::Value GetEntryValue(AZStd::string_view cvarName)
+        {
+            Dom::Value row = GetEntryRow(cvarName);
+            EXPECT_FALSE(row.IsNull());
+            return row[1].GetNodeValue();
+        }
+
+        void SetEntryValue(AZStd::string_view cvarName, Dom::Value value)
+        {
+            Dom::Value row = GetEntryRow(cvarName);
+            ASSERT_FALSE(row.IsNull());
+            auto result = Nodes::PropertyEditor::OnChanged.InvokeOnDomNode(row[1], value);
+            EXPECT_TRUE(result.IsSuccess());
+            if (!result.IsSuccess())
+            {
+                EXPECT_EQ("", result.GetError());
+            }
+        }
+
+        AZStd::unique_ptr<AZ::Console> m_console;
+        AZStd::unique_ptr<CvarAdapter> m_adapter;
+    };
+
+    TEST_F(CvarAdapterDpeTests, IntCvar)
+    {
+        EXPECT_EQ(static_cast<int32_t>(dpe_TestInt), GetEntryValue("dpe_TestInt").GetInt64());
+        SetEntryValue("dpe_TestInt", Dom::Value(42));
+        EXPECT_EQ(42, dpe_TestInt);
+        EXPECT_EQ(42, GetEntryValue("dpe_TestInt").GetInt64());
+    }
+
+    TEST_F(CvarAdapterDpeTests, UintCvar)
+    {
+        EXPECT_EQ(static_cast<uint8_t>(dpe_TestUint), GetEntryValue("dpe_TestUint").GetUint64());
```

Comment:
Just curious, why does DomValue not support GetUint8() right now?

Reply:
Dom::Value only supports getters for types it explicitly supports, so int64_t, uint64_t, and double to avoid making its API too big - it's completely safe to use this for comparison, but tells the DOM consumer that it's up to them to do a narrowing cast if needed (instead of just implicitly doing `aznumeric_cast` or something).